src/page-multi-facility/ScenarioLibraryModal.tsx

```diff
@@ -368,6 +353,25 @@ const ScenarioLibraryModal: React.FC<Props> = ({ trigger }) => {
             />
           </div>
         )}
+        <ModalDialog
```

Comment:
Just moving this confirmation dialog outside of the loop so that it doesn't get created multiple times